modules/pubxaiAnalyticsAdapter.js

```diff
@@ -235,65 +260,50 @@ export const getDeviceType = () => {
     /ipad|android 3.0|xoom|sch-i800|playbook|tablet|kindle/i.test(
       navigator.userAgent.toLowerCase()
     )
-  ) return 'tablet';
+  ) return deviceTypes.TABLET;
   if (
     /iphone|ipod|android|blackberry|opera|mini|windows\sce|palm|smartphone|iemobile/i.test(
       navigator.userAgent.toLowerCase()
     )
-  ) return 'mobile';
-  return 'desktop';
+  ) return deviceTypes.MOBILE;
+  return deviceTypes.DESKTOP;
 };
 
 /**
  * Get the approximate browser type from the user agent (or vendor if available)
  * @returns {string}
  */
 export const getBrowser = () => {
-  if (/Edg/.test(navigator.userAgent)) return 'Microsoft Edge';
+  if (/Edg/.test(navigator.userAgent)) return browserTypes.EDGE;
   else if (
     /Chrome/.test(navigator.userAgent) &&
     /Google Inc/.test(navigator.vendor)
-  ) {
-    return 'Chrome';
-  } else if (navigator.userAgent.match('CriOS')) return 'Chrome';
-  else if (/Firefox/.test(navigator.userAgent)) return 'Firefox';
+  ) return browserTypes.CHROME;
+  else if (navigator.userAgent.match('CriOS')) return browserTypes.CHROME;
+  else if (/Firefox/.test(navigator.userAgent)) return browserTypes.FIREFOX;
   else if (
     /Safari/.test(navigator.userAgent) &&
     /Apple Computer/.test(navigator.vendor)
-  ) {
-    return 'Safari';
-  } else if (
+  ) return browserTypes.SAFARI
+  else if (
     /Trident/.test(navigator.userAgent) ||
     /MSIE/.test(navigator.userAgent)
-  ) {
-    return 'Internet Explorer';
-  } else return 'Others';
+  ) return browserTypes.INTERNET_EXPLORER
+  else return browserTypes.OTHER;
 };
 
 /**
  * Get the approximate OS from the user agent (or app version, if available)
  * @returns {string}
  */
 export const getOS = () => {
-  if (navigator.userAgent.indexOf('Android') != -1) {
-    return 'Android';
-  }
-  if (navigator.userAgent.indexOf('like Mac') != -1) {
-    return 'iOS';
-  }
-  if (navigator.userAgent.indexOf('Win') != -1) {
-    return 'Windows';
-  }
-  if (navigator.userAgent.indexOf('Mac') != -1) {
-    return 'Macintosh';
-  }
-  if (navigator.userAgent.indexOf('Linux') != -1) {
-    return 'Linux';
-  }
-  if (navigator.appVersion.indexOf('X11') != -1) {
-    return 'Unix';
-  }
-  return 'Others';
+  if (navigator.userAgent.indexOf('Android') != -1) return osTypes.ANDROID
```

Comment:
this appears to be an attempt to fool the duplication check?

Reply:
@patmmccann Almost all the analytics adapters have their own implementation of browser, platform calculation. We decide to use enum not to fool the duplicates check but to make this enum specific to us as well as to reduce the payload size.

Reply:
Implementing import is beyond the scope of this PR, to be honest. If someone else changes the enum numbers or the values of the platform/browser etc, it will affect our data collection. So, we are keeping the platform/OS enums specific to our Analytics adapter only and don't want to introduce a new dependency on a library import atm.

Reply:
This isn't negotiable, members to your shower are now frozen

Reply:
Since we're having trouble getting this pr merged, please submit an independent pr for your storage fix or your adapter risks unplanned changes to enforce the rules

Reply:
Fyi, this may merge https://github.com/prebid/Prebid.js/pull/11830 if you don't work towards getting this pr merged or so an independent pr on just this fix